src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -112,21 +114,209 @@ private static IDictionary GetEnvironmentVariablesCore(EnvironmentVariableTarget
 
         private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
         {
-            string home = GetEnvironmentVariable("HOME");
+            // Get the path for the SpecialFolder
+            string path = GetFolderPathCoreWithoutValidation(folder);
+            Debug.Assert(path != null);
+
+            // If we didn't get one, or if we got one but we're not supposed to verify it,
+            // or if we're supposed to verify it and it passes verification, return the path.
+            if (path.Length == 0 ||
+                option == SpecialFolderOption.DoNotVerify ||
+                Interop.Sys.Access(path, Interop.Sys.AccessMode.R_OK) == 0)
+            {
+                return path;
+            }
+
+            // Failed verification.  If None, then we're supposed to return an empty string.
+            // If Create, we're supposed to create it and then return the path.
+            if (option == SpecialFolderOption.None)
+            {
+                return string.Empty;
+            }
+            else
+            {
+                Debug.Assert(option == SpecialFolderOption.Create);
+
+                // TODO #11151: Replace with Directory.CreateDirectory once we have access to System.IO.FileSystem here.
+                Type dirType = Type.GetType("System.IO.Directory, System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                dirType?.GetTypeInfo().GetDeclaredMethod("CreateDirectory").Invoke(null, new object[] { path });
```

Comment:
I'm not sure what Windows does in case it's unable to create the directory.  I'll try to find out.